File: app/src/main/java/com/vsa/filmoteca/data/usecase/ClearCacheUseCase.java
package com.vsa.filmoteca.data.usecase;

import com.vsa.filmoteca.data.repository.CacheRepository;

/**
 * Created by albertovecinasanchez on 18/7/16.
 */

public class ClearCacheUseCase {

    private CacheRepository mCacheRepository;

    public ClearCacheUseCase(CacheRepository cacheRepository) {
        mCacheRepository = cacheRepository;
    }

    public void clearExpiredCacheFiles() {
        mCacheRepository.clearExpiredCacheFiles();
    }

}

File: app/src/main/java/com/vsa/filmoteca/data/usecase/ClearCacheUseCase.kt
package com.vsa.filmoteca.data.usecase

import com.vsa.filmoteca.data.repository.CacheRepository
import javax.inject.Inject

/**
 * Created by albertovecinasanchez on 18/7/16.
 */
class ClearCacheUseCase @Inject constructor(private val repository: CacheRepository) {

    fun clearExpiredCacheFiles() {
        repository.clearExpiredCacheFilesAsync()
    }

}
